internal/watcher/watcher.go

```diff
@@ -74,7 +74,7 @@ type Service interface {
 	ByName(name string) (Watcher, error)
 
 	// Download blob given an event.
-	Download(ctx context.Context, w io.Writer, watcherName, key string) error
+	Download(ctx context.Context, dest, watcherName, key string) error
```

Comment:
> Change watcher server Download() method so it accepts a destination filepath (string) instead of a file writer to allow testing without needing to pass an open file writer.

`io.Writer` is nice because it keeps the function decoupled from the file system - its sole responsibility is to write data, and it's up to the caller to do the rest. It's nice that these are all possible values we can pass: *os.File, *bytes.Buffer, *blob.Writer...

I suggest to maintain it. When it comes to TestDownloadActivity, this is how it coud look like: https://gist.github.com/sevein/e7529e067601dfa1a2a9c44adf113e4b. The trick used in `gomock.AssignableToTypeOf((*os.File)(nil))` wasn't obvious to me, I'm wondering if that was the original motivation for this change.

Reply:
@sevein very nice! I was struggling with how to test that download actually writes data to the target file, so thank you for demonstrating how to do that. :)